x/service/keeper/msg_server_add_service.go

```diff
@@ -23,14 +24,57 @@ func (k msgServer) AddService(
 	logger := k.Logger(ctx).With("method", "AddService")
 	logger.Info(fmt.Sprintf("About to add a new service with msg: %v", msg))
 
+	// Validate the message.
 	if err := msg.ValidateBasic(); err != nil {
 		logger.Error(fmt.Sprintf("Adding service failed basic validation: %v", err))
 		return nil, err
 	}
 
+	// Check if the service already exists or not.
 	if _, found := k.GetService(ctx, msg.Service.Id); found {
 		logger.Error(fmt.Sprintf("Service already exists: %v", msg.Service))
-		return nil, types.ErrServiceAlreadyExists
+		return nil, types.ErrServiceAlreadyExists.Wrapf(
+			"duplicate ID: %s", msg.Service.Id,
+		)
+	}
+
+	// Retrieve the address of the actor adding the service.
+	accAddr, err := sdk.AccAddressFromBech32(msg.Address)
```

Comment:
_Not actionable, just a question._

I looked at the docs in [1], and some of the underlying code. Do you know if we are guaranteed that the signature is validated by the time `msgServer` receives it?

[1] https://docs.cosmos.network/main/build/building-modules/protobuf-annotations#signer